### DISCUSSION TITLE
Where do you keep your DB credentials?

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#6063] [Q&A]
We are starting the process of moving from mirth appliances to using mirth connect in Azure. Currently we keep our db passwords (that are used in mirth channels) in a file saved in the /folders/mirthconnect/conf/custom  folder and read\save them to global scripts on channel deployment.

Since we won't have the appliance any longer, what is the best practice for storing credentials for mirth to access? It is unclear if we will have access to the linux box hosting MC in Azure to store a file.

**Comment by jonbartels** [comment]
The typical solution is the MC config map. I strongly suggest using a config map backed by the DB, this is set in mirth.properties.

You can also use a file backed MC config map, then manipulate the `configurationmap.properties` file and it will load when MC starts.

The next option are environment variables. These can be passed from a secret store to your MC container in Azure. Then you can modify your existing global scripts to just read env vars and load to your global maps.
